phpBB/phpbb/storage/provider/provider_interface.php

```
@@ -28,4 +28,11 @@ public function get_adapter_class();
 	 * @return array	Configuration keys
 	 */
 	public function get_options();
+
+	/**
+	 * Return true if the adapter is available.
+	 *
+	 * @return bool
+	 */
+	public function is_available();
```

you should use it in the factory (throw an exception if not available)